includes/nets-easy-functions.php

```diff
@@ -449,4 +449,30 @@ function nexi_get_payment_method_title( $order, $method, $type ) {
  */
 function nexi_is_embedded( $checkout_flow ) {
 	return in_array( $checkout_flow, array( 'embedded', 'inline' ), true );
+}
+
+/**
+ * Terminate Nexi session.
+ *
+ * @param string $payment_id The payment ID.
+ * @return void
+ */
+function nexi_terminate_session( $payment_id ) {
+	if ( empty( $payment_id ) && isset( WC()->session ) && method_exists( WC()->session, 'get' ) ) {
+		$payment_id = WC()->session->get( 'dibs_payment_id' );
+	}
+
+	if ( empty( $payment_id ) ) {
+		Nets_Easy_Logger::log( 'No payment ID provided for session termination.' );
+		return;
+	}
+
+	try {
+		$response = Nets_Easy()->api->terminate_nets_easy_session( $payment_id );
+		if ( is_wp_error( $response ) ) {
+			Nets_Easy_Logger::log( 'Error terminating Nexi session: ' . $response->get_error_message() );
+		}
+	} catch ( Exception $e ) {
+		Nets_Easy_Logger::log( 'Exception when terminating Nexi session: ' . $e->getMessage() );
+	}
```

Comment:
Suggestion: I would suggest wrapping the entire function in the try catch instead of just the request. That way if we add extra things here in the future we can be sure that this wont cause any errors.